components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/internal/action/PreUpdateProfileActionExecutor.java

```diff
@@ -129,16 +123,14 @@ public void execute(User user, Map<String, String> userClaimsExcludingMultiValue
 
         ActionExecutorService actionExecutorService = SCIMCommonComponentHolder.getActionExecutorService();
 
-        if (!actionExecutorService.isExecutionEnabled(ActionType.PRE_UPDATE_PROFILE) || !isExecutable(
-                userClaimsExcludingMultiValuedClaimsToBeModified, userClaimsExcludingMultiValuedClaimsToBeDeleted,
-                simpleMultiValuedClaimsToBeAdded, simpleMultiValuedClaimsToBeRemoved)) {
+        if (!actionExecutorService.isExecutionEnabled(ActionType.PRE_UPDATE_PROFILE)) {
             return;
         }
 
         LOG.debug("Executing pre update profile action for user: " + user.getId());
 
         try {
-            Map<String, String> multiValuedClaimsToModify = getSimpleMultiValuedClaimsToModify(existingClaimsOfUser,
+            Map<String, String> multiValuedClaimsToModify = SCIMCommonUtils.getSimpleMultiValuedClaimsToModify(existingClaimsOfUser,
```

Comment:
Isn't this logic now brought in to the upper layer ?
https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/659/files#diff-73a583deac4360e8a5cff771ed84fa42f535a6deaace88053b8f039b2e6c26ecR5670-R5672